src/main/java/com/glencoesoftware/omero/zarr/OmeroAmazonS3ClientFactory.java

```diff
@@ -80,16 +106,47 @@ private String getBucketFromUri(URI uri) {
         return path.substring(0, path.indexOf("/"));
     }
 
+    /**
+     * Retrieves the region from a given URI.
+     * @param uri The URI to handle
+     * @return The region
+     */
     private String getRegionFromUri(URI uri) {
         String host = uri.getHost();
-        return host.split("\\.")[1];
+        if (host.contains("amazonaws.com")) {
+            String[] values = host.split("\\.");
+            if (values.length == 3) {
+                return Regions.DEFAULT_REGION.getName();
+            } else if (values.length > 3) {
+                return values[1];
+            }
+        }
+
+        return Regions.DEFAULT_REGION.getName();
+    }
+
+    /**
+     * Retrieves the endpoint from a given URI.
+     * @param uri The URI to handle
+     * @return The endpoint
+     */
+    private String getEndPointFromUri(URI uri) {
+        String host = uri.getHost();
+        if (host.contains("amazonaws.com")) {
+            return ENDPOINT;
+        }
+        // Check if is an endpoint other than s3.amazonaws.com
+        String[] values = host.split("\\.");
+        if (values.length > 1) {
```

Comment:
What is the use case for this additional check and the fallback to the AWS endpoint?

Reply:
I had initially in mind to support entry like ``s3://bucket/key`` i.e. no endpoint specified.
but this could come later